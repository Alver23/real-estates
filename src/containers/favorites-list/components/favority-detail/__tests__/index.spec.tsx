// Dependencies
import { shallow, ShallowWrapper } from 'enzyme';

// Under test file
import FavorityDetail from '../index';

describe('<FavorityDetail />', () => {
  let component: ShallowWrapper;
  const props = {
    images: [],
    name: 'faje name',
  };

  beforeEach(() => {
    component = shallow(<FavorityDetail {...props} />);
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const newProps = {
      ...props,
      images: [{ image: 'http://localhost/img.png', id: 1, alt: 'fake alt' }],
    };

    component.setProps({ ...newProps });
    expect(component.find('span')).toHaveLength(2);
  });
});
