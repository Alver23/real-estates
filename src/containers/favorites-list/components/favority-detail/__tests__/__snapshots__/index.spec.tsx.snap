// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavorityDetail /> should save a snapshot of the component 1`] = `
<Fragment>
  <CardImages
    images={Array []}
  />
  <div
    className="mt-4 flex flex-col"
  >
    <span
      className="text-lg text-primary font-semibold"
    >
      faje name
    </span>
    <span
      className="text-green-400 mt-2"
    >
      0
       
      propiedades guardadas
    </span>
  </div>
</Fragment>
`;
